packages/markdown-support/src/rehype-collect-headers.ts

```diff
@@ -1,32 +1,40 @@
 import { visit } from 'unist-util-visit';
+import type { Node } from 'unist';
+import type { Element } from 'hast';
 import slugger from 'github-slugger';
 
 /**  */
 export default function createCollectHeaders() {
   const headers: any[] = [];
 
-  const visitor = (node: any) => {
-    if (node.type !== 'element') return;
-    const { tagName, children } = node;
-    if (tagName[0] !== 'h') return;
-    let [_, depth] = tagName.match(/h([0-6])/) ?? [];
-    if (!depth) return;
-    depth = Number.parseInt(depth);
-
-    let text = '';
-
-    visit(node, 'text', (child) => {
-      text += (child as any).value;
-    });
-
-    let slug = node.properties.id || slugger.slug(text);
-
-    node.properties = node.properties || {};
-    node.properties.id = slug;
-    headers.push({ depth, slug, text });
-
-    return node;
+  function rehypeCollectHeaders() {
+    return function (tree: Node) {
```

Comment:
`Root` from `hast` should strongly help the rest of this function

Reply:
Done in 3392ebb